docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,201 @@
+- Enable Metrics. Boolean. Use Prometheus exporter.
+- Replicas. Number. Number of replicas.
+
+Note that we should be able to represent any parameter in a generic way. We are just defining a set of well-known parameters to be able to render a richer view.
+
+### JSON Schema Definition
+
+Based on the above parameters, this may be an example of a JSON Schema for WordPress:
+
+```json
+{ 
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+
+  "properties": {
+    "wordpressUsername": { "type": "string", "title": "Username" },
+    "wordpressPassword": { "type": "string", "title": "Password" },
+    "wordpressEmail": { "type": "string", "title": "Email" },
+    "mariadb": { 
+      "type": "object",
+      "properties": {
+        "enabled": { "type": "boolean", "title": "Enable External Database" }
+      }
+    },
+    "externalDatabase": {
+      "type": "object",
+      "title": "External Database Details",
+      "properties": {
+        "host": { "type": "string" },
+        "user": { "type": "string" },
+        "password": { "type": "string" },
+        "database": { "type": "string" },
+        "port": { "type": "integer" }
+      }
+    },
+    "persistence": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "title": "Disk Size",
+              "anyOf": [
+            {"pattern": ".*"},
+            {"enum": ["10Gi", "50Gi", "100Gi"]},
+          ]
+        }
+      }
+    },
+    "resources": {
+      "type": "object",
+      "title": "Required Resources",
+      "description": "Deployment resource requests",
+      "properties": {
+        "requests": {
+          "type": "object",
+          "properties": {
+            "memory": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["128Mi", "512Mi", "1Gi"] }
+              ]
+            },
+            "cpu": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["100m", "300m", "500m"] }
+              ]
+            }
+          }
+        },
+      }
+    },
+    "enableMetrics": { "type": "boolean", "title": "Enable Metrics" },
+    "replicas": { "type": "integer", "title": "Number of Replicas"}
+  },
+
+  "required": ["wordpressUsername", "wordpressEmail", "persistence", "resources"],
+}
+```
+
+This `values.schema.json` is a valid JSON schema for the `values.yaml` of WordPress. It can be more complete since it's possible to include every single parameter of the values.yaml file.
+
+In order to be able to provide a richer presentation of the form we are including the file `values.form.json`. This allows us to know which parameters we need to represent in the form and how to map them to valid values. In case there is a parameter that is not known we should be able to parse it as a generic input.
```

Comment:
> allows us to know which parameters we need to represent in the form

See above comments (ie. we could do this in the schema? Not sure why we wouldn't)

> and how to map them to valid values

We'd get this for free I think? I haven't checked the auto form generation from the schema, but I assume it creates input elements with names like `mariadb.enabled` rather than needing to map to other names? But again, haven't checked.